cnxarchive/views.py

```diff
@@ -525,6 +525,137 @@ def get_export(request):
     return resp
 
 
+@view_config(route_name='in-book-search', request_method='GET')
+def in_book_search(request):
+    """ Full text, in-book search """
+    results = {}
+
+    args = request.matchdict
+    ident_hash = args['ident_hash']
+    args['ident_hash'] = args['ident_hash'].split('@')[0]
+
+    try:
+        args['search_term'] = request.params.get('q', '')
+    except (TypeError, ValueError, IndexError):
+        args['search_term'] = None
+
+    id, version = split_ident_hash(ident_hash)
+    if version:
+        if '.' in version:
+            args['major_version'], \
+                args['minor_version'] = [int(v) for v in version.split('.')]
+        else:
+            args['major_version'] = int(version)
+            args['minor_version'] = None
+
+    settings = get_current_registry().settings
+    connection_string = settings[config.CONNECTION_STRING]
+    with psycopg2.connect(connection_string) as db_connection:
+        with db_connection.cursor() as cursor:
+            if not version:
+                redirect_to_latest(cursor, id, 'in-book-search',
+                                   route_args=request.params.copy())
+
+            sql_file = os.path.join(SQL_DIRECTORY, 'get-in-book-search.sql')
+            with open(sql_file, 'r') as fp:
```

Comment:
move the open/read code into database.py, so we only read it once on startup, not every search